### ISSUE TITLE
Typst fail to render label in filter block

### ISSUE THREAD

**Issue by stevengogogo** [quarto-dev/quarto-cli#8539]
### Bug description

The lemma label in filter box cannot be tracked by Typst

### Steps to reproduce

```qmd
---
title: "Typst tutorial"
date: today
format: typst
---

# Intro
 
:::{#lem-a}
## theory

$a$
:::

@lem-a

```

### Expected behavior

A PDE file with lemma 1 rendered.

### Actual behavior

```
[typst]: Compiling test.typ to test.pdf...error: cannot reference block
    ┌─ test.typ:253:0
    │
253 │ @lem-a
    │ ^^^^^^

```

### Your environment

- IDE: VScode1.85.2 MacOS: Sonoma14.2.1

### Quarto check output

```bash
Quarto 1.4.549
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.69.5: OK
      Deno version 1.37.2: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.4.549
      Path: /Applications/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: v2023.06
      Chromium: 869685

[✓] Checking LaTeX....................OK
      Using: TinyTex
      Path: /Users/stevenchiu/Library/TinyTeX/bin/universal-darwin
      Version: 2023

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.11.2
      Path: /usr/local/opt/python@3.11/bin/python3.11
      Jupyter: 5.3.0
      Kernels: probnum, python3, julia-1.8, r, jaxmetal, tfjax, gpjax, xpython, xpython-raw

(|) Checking Jupyter engine render....Traceback (most recent call last):
  File "/Applications/quarto/share/jupyter/jupyter.py", line 21, in <module>
    from notebook import notebook_execute, RestartKernel
  File "/Applications/quarto/share/jupyter/notebook.py", line 14, in <module>
    from yaml import safe_load
ModuleNotFoundError: No module named 'yaml'
[✓] Checking Jupyter engine render....OK
```

**Comment by cscheid** [comment]
Theorems/lemmas are not supported in typst yet. (They will be, soon!)

**Comment by gordonwoodhull** [comment]
Now available on `main`, implemented in #8825
